src/core/manager.c

```diff
@@ -1940,17 +1940,31 @@ static int manager_dispatch_signal_fd(sd_event_source *source, int fd, uint32_t
                         if (MANAGER_IS_SYSTEM(m)) {
```

Comment:
This deserves to be its own function. Nesting a switch inside of a switch inside of a loop is a bit too much ;)